exonum-java-binding-core/src/main/java/com/exonum/binding/storage/indices/ProofMapIndexProxy.java

```diff
@@ -210,6 +212,30 @@ public UncheckedMapProof getProof(K key) {
 
   private native UncheckedMapProof nativeGetProof(long nativeHandle, byte[] key);
 
+  /**
+   * Returns a proof that there are values mapped to the specified keys or that there are no such
+   * mappings.
+   *
+   * @param keyList list of proof map keys which might be mapped to some value, each must be 32-byte
+   *     long
+   * @throws IllegalStateException if this map is not valid
+   * @throws IllegalArgumentException if the size of any of the keys is not 32 bytes
+   */
+  public UncheckedMapProof getProof(List<K> keyList) {
+    return nativeGetMultiProof(getNativeHandle(), mergeKeysIntoByteArray(keyList));
+  }
+
+  private byte[] mergeKeysIntoByteArray(List<K> keyList) {
+    byte[] keyArray = new byte[keyList.size() * PROOF_MAP_KEY_SIZE];
```

Comment:
how about stream? )
```
    int arraySize = keyList.size() * PROOF_MAP_KEY_SIZE;
    return keyList.stream()
        .map(keySerializer::toBytes)
        .collect(Collector.of(
            () -> ByteBuffer.allocateDirect(arraySize),
            ByteBuffer::put, (a, b) -> b, ByteBuffer::array));
```

Reply:
Thanks! Only `ByteBuffer.allocate(arraySize)`, otherwise `UnsupportedOperationException` is thrown in `ByteBuffer::array`